msteams-platform/apps-in-teams-meetings/build-apps-for-anonymous-user.md

```diff
@@ -0,0 +1,229 @@
+  "timestamp": "<<timestamp>>", 
+  "id": "<<GUID2>>", 
+  "channelId": "msteams", 
+  "serviceUrl": "<<serviceURL>>", 
+  "from": { 
+    "id": "<<GUID3>>" 
+  }, 
+  "conversation": { 
+    "isGroup": true, 
+    "tenantId": "<<tenant id>>", 
+    "id": "<<chat id>>" 
+  }, 
+  "recipient": { 
+    "id": "<<bot id>>", 
+    "name": "bot name" 
+  }, 
+  "channelData": { 
+    "tenant": { 
+      "id": "<<tenant id>>" 
+    }, 
+    "source": null, 
+    "meeting": { 
+      "id": "<<meeting id>>" 
+    } 
+  } 
+} 
+```
+
+> [!NOTE]
+>
+> * When an anonymous user is added to the meeting, `membersAdded` payload object doesn't have `aadObjectId` field.
+> * When an anonymous user is added to the meeting, `from` object in the payload always have the id of the meeting organizer, even if the anonymous user was added by another presenter.
+
+**ConversationUpdate Event – MembersRemoved Payload object**
+
+```csharp
+{
+  "membersRemoved": [
+    {
+      "id": "<<GUID1>>" 
+    }
+  ],
+  "type": "conversationUpdate",
+  "timestamp": "<<timestamp>>",
+  "id": "<<GUID2>>",
+  "channelId": "msteams",
+  "serviceUrl": "<<serviceURL>>",
+  "from": {
+    "id": "<<GUID3>>"
+  },
+  "conversation": {
+    "isGroup": true,
+    "tenantId": "<<tenant id>>",
+    "id": "<<chat id>>"
+  },
+  "recipient": {
+    "id": "<<bot id>>",
+    "name": "bot name"
+  },
+  "channelData": {
+    "tenant": {
+      "id": "<<tenant id>>"
+    },
+    "source": null,
+    "meeting": {
+      "id": "<<meeting id>>"
+    }
+  }
+}
+```
+
+> [!NOTE]
+>
+> * When an anonymous user is removed from a meeting, `membersRemoved` payload object doesn't have `aadObjectId` field.
+> * When an anonymous user is removed from a meeting, `from` object in the payload always have the id of the meeting organizer, even if the anonymous user was removed by another presenter.
+
+**Create Conversation API**
+
+Create conversation API is not supported for anonymous users. The anonymous user receives the following error payload:
+
+```csharp
+{ 
+  "error": {
+    "code": "BadArgument",
+    "message": "Bot cannot create a conversation with an anonymous user"
+  }
+} 
+```
+
+**Invoke**
+
+Anonymous users can interact with Adaptive Cards, which use invoke activities.
+
+When a user selects a button on the Adaptive Card in meeting chat, the invoke request is triggered to your bot with the corresponding data through the `Action.Submit` function of Adaptive Card. The Adaptive Card data is available through the data property of the request. You receive either of the following responses to your request:
+
+* An HTTP status code `200` response with no body. An empty `200` response results in no action taken by the client.
+* The standard `200` continue response. A continue response triggers the client to update the rendered Adaptive Card with the Adaptive Cards provided in the cards array of the continue response.
+
+The example shows invoke activities request payload on submit event.
+
+```json
```

Comment:
This is the payload that the Teams client sends, but this is not the payload that the bot actually sees.